gitops-v2/new/push.yaml

```diff
@@ -71,38 +82,85 @@ jobs:
           targetPath: "$(Pipeline.Workspace)"
 
       - task: AzureCLI@2
-        displayName: "Push image to ACR"
+        displayName: "Get ACR login info"
         inputs:
           azureSubscription: ${{ parameters.azureSubscription }}
           addSpnToEnvironment: true
           workingDirectory: "."
           scriptLocation: inlineScript
           scriptType: bash
           inlineScript: |
-            set -ex
+            set -e
 
-            WORKSPACE=$(Pipeline.Workspace)
             SERVICE_NAME=$(resources.triggeringAlias)
             IMAGE_ID=$(imageId)
             IMAGE_PATH_PREFIX=${{ parameters.imagePathPrefix }}
             ACR_NAME=${{ parameters.acrName }}
-            EXTRA_TAG=${{ parameters.extraTag }}
-
-            ARTIFACT_PATH="${WORKSPACE}/${SERVICE_NAME}/${ARTIFACT_FILE_NAME}"
-            if [[ ! -f "${ARTIFACT_PATH}" ]]
-            then
-                echo "Missing artifact file ($ARTIFACT_PATH) from CI pipeline, did you trigger this pipeline manually?"
-                exit 1
-            fi
+
             IMAGE_ID_FULL="${ACR_NAME}.azurecr.io/${IMAGE_PATH_PREFIX}/${IMAGE_ID}"
             az acr login -n "${ACR_NAME}"
-            docker load < $ARTIFACT_PATH
+        condition: ne('${{ parameters.azureSubscription }}', '')
+
+      - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+        displayName: "Get ECR login info"
+        inputs:
+          workingDirectory: "."
+          awsCredentials: ${{ parameters.awsServiceConnection }}
+          regionName: ${{ parameters.awsRegion }}
+          scriptType: inline
+          inlineScript: |
+            set -e
+
+            aws ecr describe-repositories --repository-names ${SERVICE_NAME} || aws ecr create-repository --repository-name ${SERVICE_NAME} --image-scanning-configuration scanOnPush=true
+            ECR_URI=$(aws ecr describe-repositories --repository-names ${SERVICE_NAME} | jq -r '.repositories[0].repositoryUri')
+            ECR_PASSWORD=$(aws ecr get-login-password --region ${AWS_REGION})
+            docker login -u AWS -p $ECR_PASSWORD $ECR_URI
+
+            echo "##vso[task.setvariable variable=server]$ECR_URI"
+        env:
+          SERVICE_NAME: $(resources.triggeringAlias)
+        condition: ne('${{ parameters.awsRegion }}', '')
+
+      - bash: |
+          set -e
+          WORKSPACE=$(Pipeline.Workspace)
+          EXTRA_TAG=${{ parameters.extraTag }}
+          SHA_SHORT=$(shaShort)
+          SERVER=$(server)
+          ACR_NAME=${{ parameters.acrName }}
+          IMAGE_PATH_PREFIX=${{ parameters.imagePathPrefix }}
```

Comment:
While we are at it, I think we should move these to the task ENV instead. No need to do it inside of the task from what I see here.